holoviews/plotting/bokeh/callbacks.py

```diff
@@ -229,39 +229,59 @@ class Callback(object):
     def __init__(self, plot, streams, source, **params):
         self.plot = plot
         self.streams = streams
-        self.comm = self._comm_type(plot, on_msg=self.on_msg)
+        if plot.renderer.mode != 'server':
+            self.comm = self._comm_type(plot, on_msg=self.on_msg)
         self.source = source
         self.handle_ids = defaultdict(dict)
         self.callbacks = []
+        self.plot_handles = {}
+        self._event_queue = []
 
 
     def initialize(self):
         plots = [self.plot]
         if self.plot.subplots:
             plots += list(self.plot.subplots.values())
 
-        handles = self._get_plot_handles(plots)
+        self.plot_handles = self._get_plot_handles(plots)
         requested = {}
         for h in self.models+self.extra_models:
-            if h in handles:
-                requested[h] = handles[h]
+            if h in self.plot_handles:
+                requested[h] = self.plot_handles[h]
             elif h in self.extra_models:
                 print("Warning %s could not find the %s model. "
                       "The corresponding stream may not work.")
         self.handle_ids.update(self._get_stream_handle_ids(requested))
 
+        found = []
         for plot in plots:
             for handle_name in self.models:
-                if handle_name not in handles:
+                if handle_name not in self.plot_handles:
                     warn_args = (handle_name, type(self.plot).__name__,
                                  type(self).__name__)
-                    self.warning('%s handle not found on %s, cannot'
-                                 'attach %s callback' % warn_args)
+                    print('%s handle not found on %s, cannot '
+                          'attach %s callback' % warn_args)
                     continue
-                handle = handles[handle_name]
-                js_callback = self.get_customjs(requested)
-                self.set_customjs(js_callback, handle)
-                self.callbacks.append(js_callback)
+                handle = self.plot_handles[handle_name]
+
+                # Hash the plot handle with Callback type allowing multiple
+                # callbacks on one handle to be merged
+                cb_hash = (id(handle), id(type(self)))
```

Comment:
Note that the ``id`` isn't really a hash but it is supposed to be unique...for a given Python session on a given architecture. I suspect it is ok here as I doubt this information is expected to persist in anyway between sessions.